var/spack/repos/builtin/packages/reframe/package.py

```diff
@@ -42,6 +43,7 @@ class Reframe(Package):
     depends_on("py-pygelf", when="+gelf", type="run")
     depends_on("py-sphinx", when="+docs", type="build")
     depends_on("py-sphinx-rtd-theme", when="+docs", type="build")
+    depends_on("git", when="@2.0:3.1", type="run")
```

Comment:
I assume that also `reframe@master` needs git at runtime. If that is the case then the dependency should read:
```suggestion
    depends_on("git", type="run")
```

Reply:
Ciao @alalazo, thanks for the review.
Yes, it does require at runtime, but actually the bug was fixed and one won't need `git` for ReFrame 3.2 (https://github.com/eth-cscs/reframe/issues/1464).

I can change to  
```
depends_on("git", type="run")
```
for now, but I will make the change back to 
```
depends_on("git", when="@2.0:3.1", type="run")
```
when 3.2 gets released. 
What do you think?

Reply:
Then I'd just put a comment on top of the current dependency, saying something like:
```python
# ReFrame requires git up to version 3.1, see:
# https://github.com/eth-cscs/reframe/issues/1464
```
I assumed that ReFrame was requiring `git` at runtime to e.g. clone third party repositories but it doesn't seem to be the case.